engine/installation/linux/ubuntulinux.md

```diff
@@ -116,365 +119,501 @@ packages from the Docker repository:
-NetworkManager (this might slow your network).
+#### Disable `dnsmasq` in NetworkManager
 
-1. Open the `/etc/NetworkManager/NetworkManager.conf` file for editing.
+If you prefer not to change the Docker daemon's configuration to use a specific
+IP address, follow these instructions to disable `dnsmasq` in NetworkManager.
 
-        $ sudo nano /etc/NetworkManager/NetworkManager.conf
+1.  Edit the `/etc/NetworkManager/NetworkManager.conf` file.
 
-2. Comment out the `dns=dnsmasq` line:
+2.  Comment out the `dns=dnsmasq` line by adding a `#` character to the beginning
+    of the line.
 
-        dns=dnsmasq
+    ```none
+    # dns=dnsmasq
+    ```
 
-3. Save and close the file.
+    Save and close the file.
 
-4. Restart both the NetworkManager and Docker.
+4.  Restart both NetworkManager and Docker. As an alternative, you can reboot
+    your system.
 
-        $ sudo restart network-manager
-        $ sudo restart docker
+    ```bash
+    $ sudo restart network-manager
+    $ sudo restart docker
+    ```
 
 ### Configure Docker to start on boot
 
 Ubuntu uses `systemd` as its boot and service manager `15.04` onwards and `upstart`
 for versions `14.10` and below.
 
-For `15.04` and up, to configure the `docker` daemon to start on boot, run
+#### `systemd`
+
+```bash
+$ sudo systemctl enable docker
+```
 
-    $ sudo systemctl enable docker
+#### `upstart`
 
-For `14.10` and below the above installation method automatically configures `upstart`
-to start the docker daemon on boot
+For `14.10` and below, Docker is automatically configured to start on boot using
+`upstart`.
 
 ## Upgrade Docker
 
-To install the latest version of Docker with `apt-get`:
+To install the latest version of Docker with `apt-get`. The following example
+fetches information about available versions of all system packages, then
+updates Docker if a new version is available.
 
-    $ sudo apt-get upgrade docker-engine
+```bash
+$ sudo apt-get update
+$ sudo apt-get upgrade docker-engine
+```
 
 ## Uninstallation
 
 To uninstall the Docker package:
 
-    $ sudo apt-get purge docker-engine
+```bash
+$ sudo apt-get purge docker-engine
+```
 
 To uninstall the Docker package and dependencies that are no longer needed:
 
-    $ sudo apt-get autoremove --purge docker-engine
+```bash
+$ sudo apt-get autoremove --purge docker-engine
+```
 
-The above commands will not remove images, containers, volumes, or user created
-configuration files on your host. If you wish to delete all images, containers,
-and volumes run the following command:
+Images, containers, volumes, or customized configuration files on your host are
+not automatically removed. To delete all images, containers, and volumes run the
+following command:
 
-    $ rm -rf /var/lib/docker
+```bash
+$ rm -rf /var/lib/docker
+```
 
-You must delete the user created configuration files manually.
+You must delete any customized configuration files manually.
```

Comment:
I don't think so, there may be more than one customized configuration file. "Any files" is fine grammar, just like "any file". :)